### PR TITLE
Add a method to extract the VIN from UAuthority

### PR DESCRIPTION
**Pull request by neelam-kushwah**
This commit addresses issue #56

### CODE REVIEW

**Review by tamarafischer — APPROVED** [review]

File: src/main/java/org/eclipse/uprotocol/uri/UuriUtils.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * Extracts the VIN from the given UAuthority.
+     *
+     * @param uAuthority The UAuthority object from which to extract the VIN.
+     * @return An Optional containing the extracted VIN if available, or an empty Optional if the
+     * UAuthority is null, has no name.
+     */
+    static Optional<String> extractVinFromUAuthority(UAuthority uAuthority) {
+        return Optional.ofNullable(uAuthority).filter(UAuthority::hasName).map(UAuthority::getName)
+                .map(domain -> domain.split("[.]", 0)).filter(parts -> parts.length > 0)
+                .map(parts -> parts[0].startsWith("vcu") ? parts[1] : parts[0]);
```
Comment (stevenhartley) [inline src/main/java/org/eclipse/uprotocol/uri/UuriUtils.java:46]: @neelam-kushwah we cannot put this code in the eclipse project, it is specific to GM's namespace for device/domain and not portable for other vendors that might use some other naming convention (they might only use VIN or not use VIN at all).

**Review by stevenhartley — CHANGES_REQUESTED** [review]
Need to address this problem differently

**Review by stevenhartley — CHANGES_REQUESTED** [review]
We cannot merge this as-is

**Comment by stevenhartley** [comment]
Closing the PR, it cannot be merged as is, please create a new one when ready

**Pull request closed without merging**